### PR TITLE
Default v2 PolicySpec.Types field similarly as for v1

### PR DESCRIPTION
**Pull request by neiljerram**
Default the Types field for v2 PolicySpec similarly as we do for v1 at https://github.com/projectcalico/libcalico-go/blob/master/lib/converter/policy.go#L60

### CODE REVIEW

File: lib/backend/syncersv1/updateprocessors/networkpolicyprocessor.go [hunk 1]
```diff
@@ -67,7 +67,24 @@ func convertPolicyV2ToV1Spec(spec apiv2.PolicySpec) (interface{}, error) {
 		PreDNAT:       spec.PreDNAT,
 		Types:         policyTypesAPIV2ToBackend(spec.Types),
 	}
-
+	if len(spec.Types) == 0 {
```
Comment (fasaxc) [inline lib/backend/syncersv1/updateprocessors/networkpolicyprocessor.go]: I'm not 100% sure of this.  Previously, we did this defaulting on write but this is on the syncer's read path.  Previously, on the read path, we defaulted to `[ingress, egress]`, I think.

Comment (neiljerram) [reply lib/backend/syncersv1/updateprocessors/networkpolicyprocessor.go]: Do you know where the write path for v2 is?

**Comment by fasaxc** [comment]
LGTM but probably best to get @robbrockbank's opinion on whether we're adding defaulting it he right place.  Also, do we need some UT for this?

File: lib/clientv2/globalnetworkpolicy.go [hunk 2]
```diff
@@ -40,6 +40,7 @@ type globalnetworkpolicies struct {
 // Create takes the representation of a GlobalNetworkPolicy and creates it.  Returns the stored
 // representation of the GlobalNetworkPolicy, and an error, if there is any.
 func (r globalnetworkpolicies) Create(ctx context.Context, res *apiv2.GlobalNetworkPolicy, opts options.SetOptions) (*apiv2.GlobalNetworkPolicy, error) {
+	r.defaultTypesField(res)
```
Comment (fasaxc) [inline lib/clientv2/globalnetworkpolicy.go]: What about Update()?  Do we need to do Get() too (we had that in v1)?

Comment (neiljerram) [reply lib/clientv2/globalnetworkpolicy.go]: For Get(): similarly if the object concerned was created through here.  If it's coming from an existing DB, I think the defaulting for that case happens elsewhere.

Comment (robbrockbank) [reply lib/clientv2/globalnetworkpolicy.go]: I assume someone can "undo" the change with an update no?  feels like we probably should do this in the update too.

**Pull request merged**